Source/LinqToDB/DataProvider/MultipleRowsHelper.cs

```diff
@@ -16,21 +16,27 @@ public class MultipleRowsHelper<T> : MultipleRowsHelper
 		where T : notnull
 	{
 		public MultipleRowsHelper(ITable<T> table, BulkCopyOptions options)
-			: base((DataConnection)table.DataContext, options, typeof(T))
+			: base(table.DataContext, options, typeof(T))
 		{
 			TableName = BasicBulkCopy.GetTableName(SqlBuilder, options, table);
 		}
 	}
 
 	public abstract class MultipleRowsHelper
 	{
-		protected MultipleRowsHelper(DataConnection dataConnection, BulkCopyOptions options, Type entityType)
+		protected MultipleRowsHelper(IDataContext dataConnection, BulkCopyOptions options, Type entityType)
 		{
-			DataConnection = dataConnection;
+			DataConnection = dataConnection is DataConnection dc
```

Comment:
Idea, not expectation:
```suggestion
			DataConnection = 
			  dataConnection switch
			  {
			    DataConnection dc => dc,
			    DataContext dx => dx.GetDataConnection(),
			    _ => throw new ArgumentException(...),
			  };
```